src/encoded/types/file_set.py

```diff
@@ -71,3 +81,93 @@ def meta_workflow_runs(self, request: Request) -> Union[List[str], None]:
             return result
         return
 
+    @staticmethod
+    def generate_sequencing_part(request, sequencing):
+        """ The sequencing part of the file_merge_group consists of the name of the sequencer,
+            the read type and the target read length - all required properties
+        """
+        read_type_part = sequencing.get('read_type')
+        sequencer = get_item_or_none(request, sequencing.get('sequencer')).get('identifier')
+        target_read_length = sequencing.get('target_read_length')
+        flow_cell = sequencing.get('flow_cell', 'no-flow-cell')
+        return f'{sequencer}-{read_type_part}-{target_read_length}-{flow_cell}'
+
+    @staticmethod
+    def generate_assay_part(request, library):
+        """ The library of the merge_file_group contains information on the assay
+            This basically just checks the assay code isn't a single cell type and if
+            it isn't return the identifier
+        """
+        assay = get_item_or_none(request, library.get('assay'))
+        assay_code = assay.get('code')
+        if assay_code in SINGLE_CELL_ASSAY_CODES:
+            return None
```

Comment:
Of course, the concern is that we generate incorrect group IDs, when we add a new single cell assay to the portal but forget to update `SINGLE_CELL_ASSAY_CODES` (which is likely to happen). Not sure how to best deal with that if we can't rely on search here though. A foursight check that fails when it detects a new single cell assay and reminds us that way?

Reply:
Or maybe it is better to maintain a list `NON_SINGLE_CELL_ASSAY_CODES ` and do 
```
if assay_code not in NON_SINGLE_CELL_ASSAY_CODES:
     return None
```
That way we won't generate wrong group Ids, but rather no group ID at all when we add new assay types and forget to update the list here.

Reply:
Agreed!